### PR TITLE
Changed DataSearch to add drop and responsive

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataSearch to add `drop` and `responsive` properties.

#### Where should the reviewer start?

index.d.ts

#### What testing has been done on this PR?

Added stories and a unit test.

#### How should this be manually tested?

storybook

#### Do Jest tests follow these best practices?

tests follow existing patterns

#### Do the grommet docs need to be updated?

yes

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataSearch/propTypes.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
+import PropTypes from 'prop-types';
+
 let PropType = {};
 if (process.env.NODE_ENV !== 'production') {
-  PropType = {};
+  PropType = {
+    drop: PropTypes.bool,
```
Comment (MikeKingdom) [inline src/js/components/DataSearch/propTypes.js:6]: missing responsive?

**Review by MikeKingdom — COMMENTED** [review]
Looks great. Might be missing a propType.

File: src/js/components/DataSearch/propTypes.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/DataSearch/propTypes.js:6]: good catch, done

**Review by MikeKingdom — APPROVED** [review]
Looks good

**Pull request merged**